### PR TITLE
feat: Allow SVG kebab passthroughs via map

### PR DESCRIPTION
**Pull request by cmumatt**
# Description

Related PRs: #759, #749.  Related issues: #805.

Enables passthrough SVG kebab syntax via a map from  Penrose camel case names to SVG kebab syntax.

# Implementation strategy and design decisions

This is currently implemented as an internal map extracted from the SVG standard.   Issue #805 is created to subsequently externalize this map.  Other options considered including building an "escape hatch" into the style language to allow identifiers with arbitrary characters similar to many languages.  Ideas discussed including escaping with brackets, carets, dollar signs, double quotes, etc.  There was little consensus.

This approach was selected after much discussion as it is the least intrusive and achieves most of the usability goals with the tradeoffs of (a) not currently being externalized (issue #805 opened to address this later) and (b) possibly being unclear to the user that camel case is required inside the style program and the renderer will conver the camel case properties to SVG kebab syntax. 

This change allowed further renderer code to be eliminated due to the map making it redundant.

# Checklist

- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] My code follows the style guidelines of this project (e.g.: no ESLint warnings)

# Open questions

Externalizing the map is queued in issue/feature request #805.

### CODE REVIEW

File: packages/core/src/renderer/AttrMapSvg.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  colorRendering: "color-rendering",
+  dominantBaseline: "dominant-baseline",
+  enableBackground: "enable-background",
+  fillOpacity: "fill-opacity",
+  fillRule: "fill-rule",
+  floodColor: "flood-color",
+  floodOpacity: "flood-opacity",
+  fontFamily: "font-family",
+  fontSize: "font-size",
+  fontSizeaAjust: "font-size-adjust",
```
Comment (samestep) [inline packages/core/src/renderer/AttrMapSvg.ts:24]:
```suggestion
  fontSizeAdjust: "font-size-adjust",
```

File: packages/core/src/renderer/AttrMapSvg.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  panose1: "panose-1",
+  paintOrder: "paint-order",
+  pointerEvents: "pointer-events",
+  renderingIntent: "rendering-intent",
+  shapeRendering: "shape-rendering",
+  stopColor: "stop-color",
+  stopOpacity: "stop-opacity",
+  strikethroughPosition: "strikethrough-position",
+  strikethroughThickness: "strikethrough-thickness",
+  strokeDashArray: "stroke-dasharray",
```
Comment (samestep) [inline packages/core/src/renderer/AttrMapSvg.ts:51]:
```suggestion
  strokeDasharray: "stroke-dasharray",
```

File: packages/core/src/renderer/AttrMapSvg.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  underlinePosition: "underline-position",
+  underlineThickness: "underline-thickness",
+  unicodeBidi: "unicode-bidi",
+  unicodeRange: "unicode-range",
+  unitsPerEm: "units-per-em",
+  vAlphabetic: "v-alphabetic",
+  vHanging: "v-hanging",
+  vIdeographic: "v-ideographic",
+  vMathematical: "v-mathematical",
+  vEctoreffect: "vector-effect",
```
Comment (samestep) [inline packages/core/src/renderer/AttrMapSvg.ts:71]:
```suggestion
  vectorEffect: "vector-effect",
```

**Review by samestep — CHANGES_REQUESTED** [review]
a few change suggestions for `attrMapSvg`

File: packages/core/src/renderer/AttrMapSvg.ts (hunk 2 above)
Comment (cmumatt) [reply packages/core/src/renderer/AttrMapSvg.ts:51]: Since this one is used elsewhere in the codebase, will spin this off to a different issue/PR.

Comment (samestep) [reply packages/core/src/renderer/AttrMapSvg.ts:51]: Oh! I didn't realize it was... suspicious

**Review by samestep — APPROVED** [review]
Looks good! Sorry for the high number of iterations on this

**Pull request merged**